packages/@aws-cdk/aws-stepfunctions-tasks/lib/emr/private/cluster-utils.ts

```diff
@@ -115,18 +115,46 @@ export function InstanceTypeConfigPropertyToJson(property: EmrCreateCluster.Inst
   };
 }
 
+/**
+ * Render the SpotProvisioningSpecificationProperty to JSON
+ *
+ * @param property
+ */
+export function SpotProvisioningSpecificationPropertyToJson(property: EmrCreateCluster.SpotProvisioningSpecificationProperty) {
+  return {
+    AllocationStrategy: cdk.stringToCloudFormation(property.allocationStrategy?.name),
+    BlockDurationMinutes: cdk.numberToCloudFormation(property.blockDurationMinutes),
+    TimeoutAction: cdk.stringToCloudFormation(property.timeoutAction?.valueOf()),
+    TimeoutDurationMinutes: cdk.numberToCloudFormation(property.timeoutDurationMinutes),
+  };
+}
+
+/**
+ * Render the SpotProvisioningSpecificationProperty to JSON
+ *
+ * @param property
+ */
+export function OnDemandProvisioningSpecificationPropertyToJson(property: EmrCreateCluster.OnDemandProvisioningSpecificationProperty) {
+  return {
+    AllocationStrategy: cdk.stringToCloudFormation(property.allocationStrategy?.name),
+  };
+}
+
```

Comment:
why do both of these need to be in a utils file? - they're only used in one file. the preference is to keep it local until there's a need (multiple usage sites) to move it to a common location.